lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -214,13 +231,37 @@ Tooltip for each button
 =cut
 
 has tooltips => (is => 'ro',
-    isa => 'HashRef[Str]',
-    default => sub {
-        return {
-            'Save and Retry' => marktext('Save the fixes provided and attempt to continue migration'),
-            'Cancel' => marktext('Cancel the <b>migration</b>')
-    }},
-    required => 0);
+    isa => 'Maybe[HashRef[Str]]', required => 0,
+    default => undef,   # Force initializer call
+    initializer => sub {
+        my ( $self, $value, $writer_sub_ref, $attribute_meta ) = @_;
+        for ($value) {
```

Comment:
`$value` is at best an array reference, but definitely not an array.

Please use a local value for iterating the array and explicitly iterate the array:

```perl
for my $btn (@$value) { };
```